### PR TITLE
fixes #4922 feat(nimbus): Remove old risk mitigation checklist UI/mutation

### LINKED ISSUE #4922: (BE/FE) Remove old Risk Mitigation Checklist UI/mutation
Remove the checklist input from the UI on the Overview page. Also remove `riskMitigationLink` from the FE mutation request as well as remove the field from the v5 mutation in the BE.

We want to leave it around in the experiment query and summary table for now to support experiments that have that field filled out.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1130)
┆Issue Number: EXP-1130

### PR DESCRIPTION
**Pull request by LZoog**
fixes #4922 

Blocked on: #5159, #4924

Because:
* We have new questions to be built into the UI to replace this

This commit:
* Removes the input from the UI
* Removes the checklist link from the mutation
* Removes the checklist from required for launch
* Slightly alters the summary table to show the row only if set

I created EXP-1272 / #5172 for final cleanup of this down the line.

### CODE REVIEW

File: app/experimenter/nimbus-ui/src/components/Summary/TableSummary/index.test.tsx [hunk 1]
```diff
@@ -126,7 +126,7 @@ describe("TableSummary", () => {
     });
   });
 
-  describe("renders 'Risk mitigation checklist' row as expected", () => {
+  describe("renders 'Risk mitigation checklist (old)' row as expected", () => {
```
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/components/Summary/TableSummary/index.test.tsx:129]: Hmm, thought maybe we could just remove this test altogether. But I see why it's here. Might make sense to rephrase this as "obsolete implementation" rather than "old" if only because I scratched my head for a moment

File: app/experimenter/nimbus-ui/src/lib/constants.ts [hunk 2]
```diff
@@ -14,8 +14,6 @@ export const SUBMIT_ERROR =
 export const EXTERNAL_URLS = {
   TRAINING_AND_PLANNING_DOC:
     "https://mana.mozilla.org/wiki/display/FJT/Nimbus+Onboarding",
-  RISK_MITIGATION_TEMPLATE_DOC:
```
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/lib/constants.ts:17]: I haven't looked into the other Risk Mitigation issues yet, but do we have a new version of this doc somewhere? Wondering if this link will still be useful

**Review by lmorchard — APPROVED** [review]
LGTM for what it does, though I imagine there will be collisions once the blocking issues are resolved. Do we want to wait for merge until then?

File: app/experimenter/nimbus-ui/src/components/Summary/TableSummary/index.test.tsx (hunk 1 above)
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/Summary/TableSummary/index.test.tsx:129]: Oh oops, I forgot to remove that, I originally changed the row in the Summary component to also say (old) and then chatted with Ana this morning and we decided it's not needed. I think "obsolete implementation" is better for a test descriptor anyways, I'll change that when I fix anything else. We can remove this test when #5172 is done. :)

**Comment by LZoog** [comment]
> LGTM for what it does, though I imagine there will be collisions once the blocking issues are resolved. Do we want to wait for merge until then?

Yeah, I'll wait to merge until those blockers are in. Looks like you just r+'d the PR for #4924 so it shouldn't be too long. I'll fix any conflicts once that happens. Thanks for the review 😄

File: app/experimenter/nimbus-ui/src/lib/constants.ts (hunk 2 above)
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/lib/constants.ts:17]: You've seen this by now, but we do have [new links](https://github.com/mozilla/experimenter/pull/5170/files#diff-52ccc6c28c664b4fad9c3a9af82889cc7a44bff31bfad6e4b94d1aa47c2825daR29-R34). I'll double check with Ana that we shouldn't keep this one around.

**Comment by jaredlockhart** [comment]
This should no longer be blocked 👍

File: app/tests/integration/nimbus/pages/overview.py [hunk 3]
```diff
@@ -30,12 +29,3 @@ def public_description(self):
...
         name.send_keys(f"{text}")
-
-    @property
-    def risk_mitigation(self):
-        return self.find_element(*self._risk_mitigation_locator).text
-
-    @risk_mitigation.setter
-    def risk_mitigation(self, text=None):
-        name = self.find_element(*self._risk_mitigation_locator)
-        name.send_keys(f"{text}")
```
Comment (jaredlockhart) [inline app/tests/integration/nimbus/pages/overview.py:41]: Oh good you caught the e2e changes, awesome 👍
